buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy

```diff
@@ -2785,15 +2785,70 @@ class BeamModulePlugin implements Plugin<Project> {
       // distribution tarball generated by :sdks:python:sdist.
       project.configurations { distTarBall }
 
+      // Create a task to install Beam python SDK locally before running tests on
+      // Google cloud platform. The task depends on ':sdks:python:sdist' project
+      // for the tarball file.
+      // For compatible systems (Linux amd64), a 'sdistReuseWheel' system property
+      // can be provided to re-use the wheel built in this task (by configuring
+      // `--sdk_location` pipeine option in test tasks).
       def installGcpTest = project.tasks.register('installGcpTest')  {
         dependsOn setupVirtualenv
         dependsOn ':sdks:python:sdist'
+
+        // if host system wheel compatible with dataflow runner
+        def compatibleWithDataflow = ("amd64".equalsIgnoreCase(System.getProperty("os.arch")) &&
```

Comment:
Another idea is to try to get the wheel that ci builds for us (in fact, we build wheel for every PR), but we don't upload wheels to gs://beam-wheels-staging for in-flight PRs, only for submitted commits. we could potentially change that and upload  or all commits, into a bucket with lifecycle configured. The wheels that we do upload are tagged by commit ID. Sounds like more going that route is more complexity though that just building a wheel when we can. An advantage is: we will use the same wheel we release. Maybe this will be simpler when we move everything to github actions.

Reply:
Yes this is simpler when moving to GitHub Action. The PostCommit can just add a prerequisite step of building wheels.

Currently, build wheel is a GitHub Action workflow (https://github.com/apache/beam/blob/master/.github/workflows/build_wheels.yml) and PostCommits are still Jenkins tasks and cannot make use of the former.

Reply:
entered #26049. Hold on this PR for now until that issue is done